src/Aspire.Hosting.Redis/RedisBuilderExtensions.cs

```diff
@@ -19,18 +19,129 @@ public static class RedisBuilderExtensions
     /// </summary>
     /// <remarks>
     /// The default image is "redis" and the tag is "7.2.4".
+    /// Password resource will be added to the model at publish time.
     /// </remarks>
     /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
     /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
     /// <param name="port">The host port to bind the underlying container to.</param>
     /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
-    public static IResourceBuilder<RedisResource> AddRedis(this IDistributedApplicationBuilder builder, string name, int? port = null)
+    /// <example>
+    /// Use in application host.
+    /// 
+    /// the example of adding <see cref="RedisResource"/> to application model.
+    /// <code>
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    ///
+    /// var redis = builder.AddRedis("redis",6379);
+    /// 
+    /// var api = builder.AddProject&lt;Projects.Api&gt;("api")
+    ///   .WithReference(redis);
+    ///   
+    /// builder.Build().Run(); 
+    /// </code>
+    ///
+    /// </example>
+    public static IResourceBuilder<RedisResource> AddRedis(this IDistributedApplicationBuilder builder, string name, int? port)
+    {
+        return builder.AddRedis(name, port, null);
+    }
+
+    /// <summary>
+    /// Adds a Redis container to the application model.
+    /// </summary>
+    /// <remarks>
+    /// The default image is "redis" and the tag is "7.2.4".
+    /// If a password parameter is not supplied, a password resource will be added to the model at publish time.
```

Comment:
Just curious - what password would be used in this case? And does that conflict with `password">The parameter used to provide the password for the Redis resource. If <see langword="null"/> resource will be not password protected`

Reply:
Maybe we also need a test to check that if the password parameter is not supplied then a password resource is being added to the model at publish time?